Sample admin

ToonHub opened the pull request:
I created a table with all mhq samples. For this I used the original files that resulted from the sample selection. I also created a table with assessments for which I used exports from INBOveg and fieldmap. The input data is not pushed to github but can be found in a [google drive folder](https://drive.google.com/drive/folders/11rStbOup5DFmHP8FQOe_KCOSkXJIwXx0)

There are still quite a few anomalies in the assessment data that still have to be sorted out. However, the procedures for creating the tables can already be reviewed as well as the format and the content of the tables?

florisvdh commented on src/generate_mhq_sample/10_generate_mhq_terr_sample.Rmd:
```diff
@@ -1,9 +1,35 @@
 
+The original data files which are used the scripts are stored in [this google drive folder](https://drive.google.com/open?id=11rStbOup5DFmHP8FQOe_KCOSkXJIwXx0) which can be accessed within INBO.   
 
+# Generation of table with all mhq samples 
```
samples or sampling units?

The resulting `samples` data source lists sampling units, not samples, hence may also better be renamed.

florisvdh commented on src/generate_mhq_sample/10_generate_mhq_terr_sample.Rmd:
```diff
@@ -260,46 +327,44 @@ sample_1330_da <- sample_1330_da_orig %>%
-           obs_circleplot = ifelse(obs_circleplot != "geen habitat (akker, houtkant, tuin,...)", 
-                                str_replace(obs_circleplot, " ", "_"),
-                                obs_circleplot))
+write_vc(samples, file = "mhq_terr_samples", root = "../../n2khab_mhq_data", sorting = c("type", "sample_id"), strict = FALSE)
```
Some output, added simply as a reference here

```r
> glimpse(mhq_terr_samples %>% slice(seq(1,6500,500)))
Rows: 13
Columns: 14
$ sample_id        <chr> "ZSCPQ007", "11578161_2120", "11206069_2160", "12526769_2180", "95158_2310", "28232…
$ polygon_id       <chr> "562222_v2018", "BWK335", "BWK1206", "PINK90", "41039", "44562", "47029", "32636", …
$ grts_ranking     <dbl> 61453757, 11578161, 11206069, 12526769, 95158, 282322, 6951250, 11497522, 439878, 2…
$ x                <dbl> 144113.5, 36477.6, 59261.6, 24605.6, 216893.6, 224253.6, 247613.6, 219741.6, 169501…
$ y                <dbl> 222439.9, 206702.1, 221198.1, 198446.1, 199470.1, 193902.1, 188014.1, 172014.1, 168…
$ main_type        <fct> 1330, 2120, 2160, 2180, 2310, 4010, 6120, 6410, 6430, 6510, 7140, 9120, 9160
$ type             <fct> 1330_da, 2120, 2160, 2180, 2310, 4010, 6120, 6410_mo, 6430_hf, 6510_hu, 7140_meso, …
$ phab             <dbl> 100, 15, 50, 100, 100, 100, 100, 20, 70, 100, 100, 100, 100
$ sac              <dbl> 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 1
$ year_planned     <dbl> NA, 8, 9, 9, 3, 6, 4, 7, 8, 11, 7, NA, NA
$ description_orig <chr> "100% 1130; 100% rbbmr", "80% 2130_hd; 15% 2120; 5% 2160", "50% 2160; 25% 2130_hd; …
$ legacy_site      <lgl> TRUE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, FALSE, TRUE, T…
$ certain          <lgl> TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE, TRUE
$ location_id      <chr> "61453757", "11578161", "11206069", "12526769", "95158", "282322", "6951250", "1149…
```

florisvdh replied:
Considering the variable `legacy_site`, it is not yet clear which design is involved: judgment sampling, SRS etc., and whether the considered target population covered the whole of Flanders. Those attributes may be useful for your inferences later.

florisvdh commented on src/generate_mhq_sample/20_mhq_terr_sample_admin.Rmd:
```diff
@@ -0,0 +1,1011 @@
+# Generate mhq sample administration table
```
General remark about the subparagraphs in this part (e.g. below 'Samples assigned to INBO'): while the textual explanations are reasonable, it is still difficult to follow what is actually done and what the outcome is, if the resulting (intermediate) objects are not somehow visualised (e.g. with `glimpse()`, first rows of a table etc.). While such visualisation is done to enlist anomalies, it isn't for the main data workflow. This makes it uneasy to understand what is meant with titles like '_Combine sample table with admin data_', '_Combine sample admin data and assessments data_' etc. without running code in an interactive session and inspecting results. In such cases the accompanying text focuses on specific aspects, not on explanation of the actual step (it's my impression, maybe partly wrong).

florisvdh commented on src/generate_mhq_sample/20_mhq_terr_sample_admin.Rmd:
````diff
@@ -0,0 +1,1011 @@
+           sample_source = "habitatmap")
+```
+
+## Create new samples when is_type_target = FALSE and allow_other_type = TRUE
````
```suggestion
## Create new sampling units when is_type_target = FALSE and allow_other_type = TRUE
```

florisvdh commented on src/generate_mhq_sample/20_mhq_terr_sample_admin.Rmd:
````diff
@@ -0,0 +1,1011 @@
+```
+
+
+## Add new samples to sample admin table
````
```suggestion
## Add new sampling units to sample admin table
```

florisvdh commented on src/generate_mhq_sample/20_mhq_terr_sample_admin.Rmd:
```diff
@@ -0,0 +1,1011 @@
+    mutate(n = n()) %>%
+    ungroup()
+
+write_vc(mhq_assessments_summary, root = "../../n2khab_mhq_data" , file = "mhq_assessments", sorting = c("type_target", "grts_ranking"))
```
Added selected preview as a reference for discussion:

```r
> mhq_assessments %>% slice(seq(1,1401,100)) %>% knitr::kable()
```

|sample_id         | grts_ranking|assessment_status  |type_target |is_type_target |type_observed |location_change | distance_from_sample|   x_orig|   y_orig| x_measured| y_measured|
|:-----------------|------------:|:------------------|:-----------|:--------------|:-------------|:---------------|--------------------:|--------:|--------:|----------:|----------:|
|1338805_1330_da   |      1338805|in situ assessment |1330_da     |TRUE           |1330_da       |FALSE           |                  4.9|  61661.6| 221902.1|    61666.0|   221899.9|
|1781173_1330_hpr  |      1781173|in situ assessment |1330_hpr    |FALSE          |NA            |NA              |                   NA|  62301.6| 222030.1|         NA|         NA|
|1036081_2120      |      1036081|in situ assessment |2120        |FALSE          |NA            |NA              |                   NA|  34173.6| 204462.1|         NA|         NA|
|16772913_2190_mp  |     16772913|in situ assessment |2190_mp     |TRUE           |2190_mp       |TRUE            |                  0.0|  32733.6| 203598.1|    32733.6|   203598.1|
|359574_2330_bu    |       359574|in situ assessment |2330_bu     |FALSE          |2310          |FALSE           |                  0.1| 168765.6| 205998.1|   168765.6|   205998.2|
|1762774_2330_bu   |      1762774|in situ assessment |2330_bu     |TRUE           |2330_bu       |FALSE           |                  0.1| 185661.6| 211054.1|   185661.6|   211054.1|
|120530_4030       |       120530|in situ assessment |4030        |TRUE           |4030          |FALSE           |                  0.1| 224765.6| 186926.1|   224765.6|   186926.2|
|2181769_6230      |      2181769|in situ assessment |6230        |FALSE          |NA            |NA              |                   NA| 112029.6| 193550.1|         NA|         NA|
|753630_6230_hmo   |       753630|in situ assessment |6230_hmo    |TRUE           |6230_hmo      |TRUE            |                 31.3| 160157.6| 225582.1|   160126.4|   225579.4|
|912594_6230_hn    |       912594|in situ assessment |6230_hn     |FALSE          |NA            |NA              |                   NA| 224445.6| 194670.1|         NA|         NA|
|7072966_6410_mo   |      7072966|in situ assessment |6410_mo     |TRUE           |6410_mo       |FALSE           |                  0.3| 162429.6| 179534.1|   162429.7|   179534.4|
|4599942_6410_ve   |      4599942|in situ assessment |6410_ve     |TRUE           |6410_ve       |TRUE            |                 50.3| 179069.6| 184846.1|   179068.8|   184795.8|
|1052582_6510_hu   |      1052582|in situ assessment |6510_hu     |TRUE           |6510_hu       |FALSE           |                  0.1| 185885.6| 181038.1|   185885.6|   181038.1|
|10622226_6510_hus |     10622226|in situ assessment |6510_hus    |TRUE           |6510_hus      |FALSE           |                  0.0| 219133.6| 217454.1|   219133.6|   217454.1|
|4322230_7140_meso |      4322230|in situ assessment |7140_meso   |FALSE          |NA            |NA              |                   NA| 218269.6| 198830.1|         NA|         NA|

ToonHub commented:
I changed the code to generate the tables as discussed  in [this document](https://docs.google.com/document/d/1Jsx29YVudhVFdy8t8z1wCtEc26B7yf4YMewt9pwfUgc/edit#heading=h.b4ti8bmcaazs). The data can be found in [this google drive folder](https://drive.google.com/open?id=1JpYol4Fmv_a6tjm3CBZ4RFU_81f9Wr8G). 

@florisvdh can you review the changes?

florisvdh commented:
I see there is another branch `add_data_anb`, with an extra commit (below) which is not in `sample-admin` (hence not occurring in this PR) but which also merged all commits from `sample-admin` branch (#18). Should the current PR be changed to refer to `add_data_anb` and should we not drop `sample-admin`? (Or is the extra commit in `add_data_anb` unneeded here?) I'm a bit confused about the role of the 2 branches.

```
commit e4b8d8e2736e35f0c58aa743e75eafef7728359d
Author: WESTRA <toon.westra@inbo.be>
Date:   Tue Jun 9 11:35:05 2020 +0200

    add functions to access firebird database
    add cover scales table

A       data/coverscales_mhq.tsv
A       data/coverscales_mhq.yml
A       src/generate_mhq_sample/_11_data_access_functions.Rmd
M       src/generate_mhq_sample/_17_query_fieldmap.Rmd
A       src/generate_mhq_sample/_18_query_fieldmap_fb.Rmd
```

florisvdh commented:
@ToonHub can you 'resolve conversations' of the previous review which your handled already? (i.e. by clicking 'resolve conversation'). This makes it easier to track.

ToonHub commented:
> I see there is another branch `add_data_anb`, with an extra commit (below) which is not in `sample-admin` (hence not occurring in this PR) but which also merged all commits from `sample-admin` branch (#18). Should the current PR be changed to refer to `add_data_anb` and should we not drop `sample-admin`? (Or is the extra commit in `add_data_anb` unneeded here?) I'm a bit confused about the role of the 2 branches.
> 
> ```
> commit e4b8d8e2736e35f0c58aa743e75eafef7728359d
> Author: WESTRA <toon.westra@inbo.be>
> Date:   Tue Jun 9 11:35:05 2020 +0200
> 
>     add functions to access firebird database
>     add cover scales table
> 
> A       data/coverscales_mhq.tsv
> A       data/coverscales_mhq.yml
> A       src/generate_mhq_sample/_11_data_access_functions.Rmd
> M       src/generate_mhq_sample/_17_query_fieldmap.Rmd
> A       src/generate_mhq_sample/_18_query_fieldmap_fb.Rmd
> ```

Sorry for the confusion.

I am integrating new data about ANB assessments in the work flow. I started a new branch for this because the markdown document will not build until this integration is finished. 
I tought it was better to review an intermediate version that builds properly (in the sample admin branch). This intermediate version illustrates the structure of the tables that are generated. Yet some data still have to be added to the tables (which I am doing in the 'add data anb' branch ).

florisvdh commented:
Thanks for clarifying @ToonHub.

florisvdh commented on src/generate_mhq_sample/10_generate_mhq_terr_sample.Rmd:
```diff
@@ -53,153 +56,174 @@ schemes_mhq %>%
 
-sample_dunes_orig <- read.csv2("../../n2khab_mhq_data/original/sample/sample-locations/kustduinen_hab/steekproef_duinen_tabel_versie2018-11-2018.csv",
-                               stringsAsFactors = FALSE) 
+sample_dunes_sf_orig <- read.csv2("../../mhq_sample-admin_data/original/sample/sample-locations/kustduinen_hab/steekproef_duinen_tabel_versie2020-05-28.csv",
```
Also note that the `n2khab_mhq_data` folder should be git-ignored (I suppose), which it currently isn't.

ToonHub commented:
> * I did not see a clear explanation for the reason to store two 'check_update' dataframes in the repo? If unneeded, better don't use them. If needed, explanation is needed why. At first sight, it complicates matters so I'd prefer to keep the end results as simple as possible.

It is just a temporary step to be able to compare the original file and the updated file (based on assessment information). But in the end we are just going to overwrite the file.

> * In order to ease future inspection of git history (assessing sample changes over the years), it would be more convenient to use `write_vc()` with `optimized = FALSE`. It will take more disk space, but it also means that factor levels are not replaced by integer indices, so that interpretation is more straightforward.

OK

> * Advice on `mhq_terr_refpoints_details`:
>   
>   * don't repeat the `mhq_terr_refpoints` columns except for `scheme` + `point_code` - which together (should) form a unique combination (hence my suggestion for the name 'schemepoints')
>   * best add a year or date which the details belong to (they are time-dependent for each refpoint)
>     
>     * hence `_temporal` would be a more explanatory suffix than `_details`
>   * `year_planned`: I think this is as static as the other attributes in the `mhq_terr_refpoints` table? Then it makes more sense to move `year_planned` in there.

I added year_planned to `mhq_terr_refpoints`. I also added sac and db_ref to `mhq_terr_refpoints`. Sac is needed to evaluate 'allow_other_type' and 'db_ref' is needed to make a link to the assessment data. The workflow continues from 'mhq_terr_refpoints' instead of 'mhq_terr_refpoints_details'. 
I deleted 'mhq_terr_refpoints_details' but I created a table 'mhq_terr_details_samplingframe' with details about the sampling frame: phab, certain, details_orig. This is because I do not have a standardized version of the habitatmap which was originally used yet. It is on my to do list.



> * `mhq_terr_refpoints`: some combinations of `scheme` x `refpoint` are non-unique (i.e. the below cases, which I aggregated). This is because of multiple occurring values for `type`. Are these the correct 'target' types (= the explanation you gave to `type` when introducing this table) of the considered schemes? In a [further comment](https://github.com/inbo/n2khab-sample-admin/pull/17#discussion_r439383455), I argue that it seems that `target_type` in the assessments table has another meaning, i.e. the _assessed (evaluated) type_ (irrespective of a scheme, hence can be main or subtype).
> 
> ```r
> # A tibble: 14 x 3
>    scheme type     nr_points
>    <fct>  <fct>        <int>
>  1 HQ2130 2130_had         7
>  2 HQ2130 2130_hd          7
>  3 HQ2190 2190             1
>  4 HQ2190 2190_mp          1
>  5 HQ6230 6230_ha         13
>  6 HQ6230 6230_hmo        14
>  7 HQ6230 6230_hn         17
>  8 HQ6510 6510_hu          3
>  9 HQ6510 6510_huk         3
> 10 HQ91E0 91E0_va          4
> 11 HQ91E0 91E0_vc          1
> 12 HQ91E0 91E0_vm          5
> 13 HQ91E0 91E0_vn          3
> 14 HQ91E0 91E0_vo          3
> ```
> 
> Note also that this goes along with the fact that different values of `sampling_unit_code` exist for different subtypes (of the same main type) at the _same_ point, e.g. `36917_2130_hd` and `36917_2130_had`. Is this correct? This seems a violation of the sampling design for one scheme.
> 
> Considering that 1) only main types (by that I mean: not distinguishing subtypes) are 'targetted' by the MHQ schemes (and consequently in the sample draw), and 2) the presumably different meaning of `target_type` in the assessments, I suggest to drop the `type` variable here. I don't see a clear need for it. If it must be kept, avoid speaking about 'target type' in two different contexts / meanings.
> 

There seems to be a misunderstanding. In most cases the subtype is targetted, except for very uncommen subtypes. This because we also want information on the quality of every subtype. That is why we (1) drew a sample for the main_type, (2) selected extra sampling units within sac and (3) selected extra sampling units for subtypes which were not sufficiently covered. So indeed, the combination scheme x refpoint is not unique. The same scheme x refpoint-combination can contain more than one sampling_unit, but of course only one of them can be valid.

It might be confusing that we use the main type in the scheme names. On the other hand, by using the variable 'type' it seems clear to me that the subtype is targetted.  



> * Looking forward: a workflow will be needed for updating assessments, refpoints and refpoints_validity as time passes. I don't know how you think about this for MHQ but I'd suggest to set up a repeatable workflow to maintain these datasets in the future, preferably as uniform and simple as possible (and to which co-workers will need to adapt).

work in progress

> * General note on terminology, please do a general search on this (I didn't manually mark all of them):
>   
>   * A _sample_ is a collection of _sampling units_. In several places the first is used where the second is meant.

>   * Use _sampling frame_, not sample frame
>   * Spelling: access**i**ble, not accessable. I mention this one because it also occurs misspelled in a dataset (tsv/yml).
>   * On some occasions the term 'reference point' is used, on many you use 'locations' (without clear definition I think). Are they synonyms? Then either only use 'reference point' (is most clear), or explicitly state that they are synonyms and used interchangeably.

OK

> * Note: I didn't check the actual data steps.
> * Note: I didn't focus on linguistics or typos.

Spelling checkers made me lazy

florisvdh commented:
Thanks for the clarifications. 

> This because we also want information on the quality of every subtype.

OK, then this does indeed conform to your definition _"type: the target type, i.e. the type for which the sample was drawn"_.

Is `type` in refpoints is effectively connected to `target_type` in assessments? If so, then perhaps rename as `target_type`. If not - I imagine `target_type` in _assessments_ can have broader applicability than that from the sampling design, i.e. just a type from habitatmap that is assessed at a refpoint by `is_target_type` - , then better not define `type` as 'target type' in the text on the _refpoints_ dataset (as cited above). In that case a design attribute name such as 'domain' could be a good alternative (there's an aim to obtain values of state and trend about each (sub)type). 'type' in itself is less self-explanatory.

> I added year_planned to mhq_terr_refpoints. I also added sac and db_ref to mhq_terr_refpoints.

OK. As sac can evolve, in the long term you'll probably need an extra column that defines the period in which an attribute value is valid for a reference point.

>  I deleted 'mhq_terr_refpoints_details' but I created a table 'mhq_terr_details_samplingframe' with details about the sampling frame: phab, certain, details_orig

OK

ToonHub commented on data/mhq_terr_refpoints_details_check_update.yml:
```diff
@@ -0,0 +1,384 @@
+..generic:
```
see response to general remarks below

ToonHub commented on src/generate_mhq_sample/10_generate_mhq_terr_sample.Rmd:
```diff
@@ -1,33 +1,36 @@
+- refpoint_code: a unique code for the reference point of each sampling unit; this is the point in which it is evaluated wether the target type is present or not 
+- legacy_site: TRUE/FALSE
+- xy-coordinates of the reference points in Lambert72 (crs = 31370)
+- type: the target type, i.e. the type for which the sample was drawn
```
type = type_target so I will change 'type' to 'type_target' to avoid confusion

ToonHub commented on src/generate_mhq_sample/10_generate_mhq_terr_sample.Rmd:
```diff
@@ -1,33 +1,36 @@
+- legacy_site: TRUE/FALSE
+- xy-coordinates of the reference points in Lambert72 (crs = 31370)
+- type: the target type, i.e. the type for which the sample was drawn
+- grts_ranking: ranking number of the grts master grid cell in which the sampling_unit is located
```
OK

ToonHub commented on src/generate_mhq_sample/10_generate_mhq_terr_sample.Rmd:
```diff
@@ -1,33 +1,36 @@
+    - description_orig
 
-Note that the mhq sample selection is based on an older version of the habitat map than the versions made available in [Zenodo](https://zenodo.org/record/3540576#.XoRkVYgzaUk). The sample frame which is derived from the older version of the habitat map is stored in [this google drive folder](https://drive.google.com/open?id=11rStbOup5DFmHP8FQOe_KCOSkXJIwXx0) which can be accessed within INBO.  
+Note that the mhq sample selection is based on an older version of the habitat map than the versions made available in [Zenodo](https://zenodo.org/record/3540576#.XoRkVYgzaUk). The sample frame which is derived from the older version of the habitat map is stored in [this google drive folder](https://drive.google.com/open?id=1JpYol4Fmv_a6tjm3CBZ4RFU_81f9Wr8G) which can be accessed within INBO.  
```
OK

ToonHub commented on src/generate_mhq_sample/20_mhq_terr_sample_admin.Rmd:
```diff
@@ -1,69 +1,97 @@
-In the mhq sample adminstration table we want to document assessments and measurements of the mhq samples. An assessment of a mhq sample is seen as evaluation whether or not a sample is valid. A mhq sample is valid if the target type (= the type for which the sample was selected) corresponds with the observed type in the center of the grid cell. A location can contain different samples, with each ...
+In the mhq sample adminstration table we want to document assessments and measurements of the mhq samples. 
+An assessment of a mhq sample is seen as evaluation whether or not a sample is valid. 
+A mhq sample is valid if the target type (= the type for which the sample was selected) corresponds with the observed type in the center of the grid cell. 
```
see response to general remarks

ToonHub commented on src/generate_mhq_sample/20_mhq_terr_sample_admin.Rmd:
```diff
@@ -241,35 +278,42 @@ check_plots_data_import %>%
+However it is not explicitely recorded which of the cover types is located in the center of the plot. 
+Yet this is essential information to determine if a sample is valid. 
+Therefore, when a plot contains different cover types we assume that the plot center is located in the type recorded in the square plot that corresponds to (one of) the target type(s). 
+When none of the target habitat types are observed we assume that the plot center is located in the type that belongs to the same main type as (one of) the target habitat types. We apply these rules and check of we select one type for each plot this way.
```
For example: the target types in the reference point are 4010 and 6510_hu. the observed types in the plot are 6510_hus and 6210. then we assume that 6510_hus is the observed type in the reference point as it belongs to the main type 6510, which is the main type of one of the target types.

florisvdh replied:
OK, I see what you mean now. The confusion came from "... the type that belongs to the same main type as ...". From "When none of the target habitat types are observed" it is not evident that _another subtype from a same_ (targetted) main type _is_ observed, which is the case you handle. Best describe the handled circumstance more clearly.

Do you describe what happens in the following cases then (perhaps they don't occur?), where 4010 and 6510_hu are target types?

1. 6210 and 6410_mo were observed 
2. 4010 and 6510_hu (the target types) were observed

ToonHub replied:
OK. I clarified it by using an example. 
Unfortunately, the two other cases you describe do occur (only a couple of cases). In case of 1. we select the type with the highest cover. In case of 2. we report it to the fieldwork coordinator to clarify.

florisvdh replied:
OK, best distinguish between those cases in the text

ToonHub replied:
OK I did

ToonHub commented on src/generate_mhq_sample/20_mhq_terr_sample_admin.Rmd:
```diff
@@ -933,79 +990,197 @@ check_anb_type %>%
-### Overview of all assessed samples
+### Update mhq_terr_refpoints
+
+The new sampling units/reference points are added to the file 'mhq_terr_refpoints'. For now, we give the updated file a new name: 'mhq_terr_refpoints_check_update'.   
```
see response to general remarks

ToonHub commented on src/generate_mhq_sample/20_mhq_terr_sample_admin.Rmd:
````diff
@@ -933,79 +990,197 @@ check_anb_type %>%
+
+```{r}
+mhq_terr_assessments <- sample_assessments_update2 %>%
+    select(point_code, type_target, is_type_target, type_observed, inaccessable, assessment_date = assessment_year, assessment_source, x, y)
````
OK I changed this. Now assessment_date  is a date variable. When only the assessment year is known I create a dummy date as you suggest.

florisvdh commented on src/generate_mhq_sample/20_mhq_terr_sample_admin.Rmd:
```diff
@@ -280,15 +280,15 @@ check_plots_data_import %>%
 Vegetation type cover is measured both in the square plot and the circle plot. 
 The field protocol prescribes that a plot is only recorded when the plot center is located within (one of) the target habitat type(s). 
-When the observed type is different from (one of) the target type(s), the plot can be recorded when the GRTS ranking of the sample location is smaller than the largest GRTS ranking in the set of samples for the observed type. 
+When the observed type is different from (one of) the target type(s), the sampling unit can be measured when grts_ranking_draw of the reference point is smaller than the largest rts_ranking_draw in the sample of the observed type. 
```
```suggestion
When the observed type is different from (one of) the target type(s), the sampling unit can be measured when grts_ranking_draw of the reference point is smaller than the largest grts_ranking_draw in the sample of the observed type. 
```

ToonHub replied:
OK

ToonHub commented on src/generate_mhq_sample/20_mhq_terr_sample_admin.Rmd:
```diff
@@ -241,35 +278,42 @@ check_plots_data_import %>%
 In some cases the subtype is not specified. When this is the case, we assume that the subtype of the observed type corresponds to the target type.
 
-Some recorded plots contain different habitat types or other cover types. However it is not explicitely recorded which of the cover types is located in the center of the plot. Yet this is essential information to determine if a sample is valid. Therefore, when a plot contains different cover types we assume that the plot center is located in the type recorded in the square plot that corresponds t ...
+Some recorded plots contain different habitat types or other cover types. 
```
land cover/ land use types (for example 'akker')

florisvdh replied:
OK, best clarify this in the text
